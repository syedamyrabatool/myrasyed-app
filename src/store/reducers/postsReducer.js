import { GET_POSTS, GET_POSTS_SUCCESS, GET_POSTS_FAILED } from "../types";

const initialState = {
  isLoading: false,
  posts: [],
  errorMsg: '',
};

const postsReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_POSTS:
      return {
        ...state,
        isLoading: true,
      };
    case GET_POSTS_SUCCESS:
      return {
        ...state,
        isLoading: false,
        posts: action.posts,
        errorMsg: '',
      };
    case GET_POSTS_FAILED:
      return {
        ...state,
        isLoading: false,
        posts: [],
        errorMsg: action.errorMsg,
      };
    default:
      return state;
  }
};

export default postsReducer;