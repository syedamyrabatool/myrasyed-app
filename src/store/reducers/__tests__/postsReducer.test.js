import postsReducer from "../postsReducer";

import {
  GET_POSTS,
  GET_POSTS_SUCCESS,
  GET_POSTS_FAILED
} from "../../types";

describe("reducers", () => {
  it("should return the initial state", () => {
    expect(postsReducer(undefined, {})).toEqual(
      {
        isLoading: false,
        posts: [],
        errorMsg: ''
      }
    )
    expect(postsReducer(undefined, GET_POSTS)).toMatchSnapshot();
  });
  it('should return the updated state', () => {
    expect(postsReducer([], {
      type: GET_POSTS
    })).toEqual(
      {isLoading: true}
    );

    expect(postsReducer([], {
      type: GET_POSTS_SUCCESS
    })).toEqual(
      {
        posts: undefined,
        errorMsg: '',
        isLoading: false,
      }
    );

    expect(postsReducer([], {
      type: GET_POSTS_FAILED
    })).toEqual(
      {
        posts: [],
        errorMsg: undefined,
        isLoading: false,
      }
    );
  });
})