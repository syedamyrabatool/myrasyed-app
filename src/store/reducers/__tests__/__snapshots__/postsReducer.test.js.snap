// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers should return the initial state 1`] = `
Object {
  "errorMsg": "",
  "isLoading": false,
  "posts": Array [],
}
`;
